Ensure that ports can be disabled

guusdk opened the pull request:
The previous approach, where an _empty_ value for the configuration property that controls on what TCP port the webserver will be listening, caused a NullPointerException to be thrown.

In this commit, the 'disbled' state is represented by the value `-1` (rather than a missing value) in of the corresponding system properties. In the frontend, the value is still shown as absent, as that's more likely to be understood correctly by end-users.

fixes #1

guusdk committed:
Ensure that ports can be disabled

The previous approach, where an _empty_ value for the configuration property that controls on what TCP port the webserver will be listening, caused a NullPointerException to be thrown.

In this commit, the 'disbled' state is represented by the value `-1` (rather than a missing value) in of the corresponding system properties. In the frontend, the value is still shown as absent, as that's more likely to be understood correctly by end-users.

fixes #1

Fishbowler commented on src/java/org/igniterealtime/openfire/plugins/jmxhttpbridge/web/admin/ConfigServlet.java:
```diff
@@ -130,8 +130,8 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
-        request.setAttribute("port", StatsWebServer.JMXHTTPBRIDGE_WEBSERVER_PORT.getValue());
-        request.setAttribute("secureport", StatsWebServer.JMXHTTPBRIDGE_WEBSERVER_PORT_SECURE.getValue());
+        request.setAttribute("port", StatsWebServer.JMXHTTPBRIDGE_WEBSERVER_PORT.getValue() == -1 ? null : StatsWebServer.JMXHTTPBRIDGE_WEBSERVER_PORT.getValue());
+        request.setAttribute("secureport", StatsWebServer.JMXHTTPBRIDGE_WEBSERVER_PORT_SECURE.getValue() == -1 ? null : StatsWebServer.JMXHTTPBRIDGE_WEBSERVER_PORT_SECURE.getValue());
```
Should these be `< 0` tests to be consistent with the descriptions in the i18n / javadoc?

guusdk replied:
That's sensible, thanks.

guusdk committed:
Review feedback: consistently represent 'disabled port'

Now all code consistently uses any negative number to represent a disabled state of a TCP port.

guusdk committed:
Add changelog entry reflecting fix for #1
